trireme.go

```diff
@@ -413,14 +414,23 @@ func (t *trireme) Supervisor(kind constants.PUType) supervisor.Supervisor {
 
 // run is the main function for running Trireme
 func (t *trireme) run() {
+
+	concurrency := runtime.NumCPU() - 1
+	sem := make(chan bool, concurrency)
 	for {
 		select {
 		case req := <-t.requests:
 			zap.L().Debug("Handling Trireme Request",
 				zap.Int("type", req.reqType),
 				zap.String("contextID", req.contextID),
 			)
-			req.returnChan <- t.handleRequest(req)
+
+			sem <- true
+			go func(req *triremeRequest) {
+				req.returnChan <- t.handleRequest(req)
```

Comment:
Is there any way two requests could be about the same context and cause out of order processing etc ?